README.md

```diff
@@ -217,14 +250,20 @@ We can easily queue all the jobs into the worker thread pool (see above), but th
 to continue when the work is done.  You could (but shouldn't) employ some blocking, locking, mutex +
 condition variable monstrosity, but you shouldn't.
 
+[//]: # shouldn't used reduntantly 
+
 Instead LokiMQ provides a mechanism for this by allowing you to submit a batch of jobs with a
 completion callback.  All jobs will be queued and, when the last one finishes, the finalization
 callback will be queued to continue with the task.
 
+[//]: # this sounds a lot like futures/promises, should we consider using them in the API?
+
 From the caller point of view this requires splitting the logic into two parts, a "Before" that sets
 up the batch, a "Job" that does the work (multiple times), and an "After" that continues once all
 jobs are finished.
 
+[//]: # what about timeouts / other error conditions WRT callbacks? promises handle this well...
```

Comment:
There's no possible timeout on a batch job: it's something to run in a thread, and the job either returns something or exits by exception.  Batch jobs really shouldn't be used for things that can block indefinitely because that would tie up a thread; rather the application should do something smarter (for example, having some code to shepherd the blocking operations, either in its own thread or perhaps using a lokimq periodic timer).

They are really meant for "hey I've got a bunch of computationally intensive stuff that I want to hand off to some threads and tell me when they are done."